experimental/packages/opentelemetry-exporter-metrics-otlp-http/test/browser/CollectorMetricExporter.test.ts

```diff
@@ -386,55 +392,61 @@ describe('OTLPMetricExporter - web', () => {
-  it('should override global headers config with signal headers defined via env', () => {
-    envSource.OTEL_EXPORTER_OTLP_HEADERS = 'foo=bar,bar=foo';
-    envSource.OTEL_EXPORTER_OTLP_METRICS_HEADERS = 'foo=boo';
-    const collectorExporter = new OTLPMetricExporter({ headers: {} });
-    // @ts-expect-error access internal property for testing
-    assert.strictEqual(collectorExporter._headers.foo, 'boo');
-    // @ts-expect-error access internal property for testing
-    assert.strictEqual(collectorExporter._headers.bar, 'foo');
-    envSource.OTEL_EXPORTER_OTLP_METRICS_HEADERS = '';
-    envSource.OTEL_EXPORTER_OTLP_HEADERS = '';
+  describe('when configuring via environment', () => {
+    const envSource = window as any;
+    it('should use url defined in env', () => {
+      envSource.OTEL_EXPORTER_OTLP_ENDPOINT = 'http://foo.bar/v1/metrics';
+      const collectorExporter = new OTLPMetricExporter();
+      assert.strictEqual(
+        collectorExporter._otlpExporter.url,
+        envSource.OTEL_EXPORTER_OTLP_ENDPOINT
+      );
+      envSource.OTEL_EXPORTER_OTLP_ENDPOINT = '';
+    });
+    it('should use url defined in env and append version and signal when not present', () => {
+      envSource.OTEL_EXPORTER_OTLP_ENDPOINT = 'http://foo.bar';
+      const collectorExporter = new OTLPMetricExporter();
+      assert.strictEqual(
+        collectorExporter._otlpExporter.url,
+        `${envSource.OTEL_EXPORTER_OTLP_ENDPOINT}/v1/metrics`
+      );
+      envSource.OTEL_EXPORTER_OTLP_ENDPOINT = '';
+    });
+    it('should override global exporter url with signal url defined in env', () => {
+      envSource.OTEL_EXPORTER_OTLP_ENDPOINT = 'http://foo.bar';
+      envSource.OTEL_EXPORTER_OTLP_METRICS_ENDPOINT = 'http://foo.metrics';
+      const collectorExporter = new OTLPMetricExporter();
+      assert.strictEqual(
+        collectorExporter._otlpExporter.url,
+        envSource.OTEL_EXPORTER_OTLP_METRICS_ENDPOINT
+      );
+      envSource.OTEL_EXPORTER_OTLP_ENDPOINT = '';
+      envSource.OTEL_EXPORTER_OTLP_METRICS_ENDPOINT = '';
+    });
+    it('should use headers defined via env', () => {
+      envSource.OTEL_EXPORTER_OTLP_HEADERS = 'foo=bar';
+      const collectorExporter = new OTLPMetricExporter({
+        headers: {},
+        aggregationTemporality: AggregationTemporality.CUMULATIVE
+      });
+      // @ts-expect-error access internal property for testing
+      assert.strictEqual(collectorExporter._otlpExporter._headers.foo, 'bar');
+      envSource.OTEL_EXPORTER_OTLP_HEADERS = '';
+    });
+    it('should override global headers config with signal headers defined via env', () => {
+      envSource.OTEL_EXPORTER_OTLP_HEADERS = 'foo=bar,bar=foo';
+      envSource.OTEL_EXPORTER_OTLP_METRICS_HEADERS = 'foo=boo';
+      const collectorExporter = new OTLPMetricExporter({
+        headers: {},
+        aggregationTemporality: AggregationTemporality.CUMULATIVE
+      });
+      // @ts-expect-error access internal property for testing
```

Comment:
nit: You can access private properties with `collectorExporter['_otlpExporter']` in tests (it won't work with dependencies since the type information of private properties are wiped out).

Reply:
Oh, that's good to know, thanks. :slightly_smiling_face: 
I changed it in [fcffd62](https://github.com/open-telemetry/opentelemetry-js/pull/2871/commits/fcffd628b2b97aec4d6d3702bceb0e995fd15aa8) :slightly_smiling_face: